doc/production-hosting.md

```diff
@@ -1,40 +1,231 @@
+Cloudflare.
+
+Any maintainer can push to `gh-pages` though in practice this needs to be kept
+in sync with the production deploy.
+
+[github pages]: https://pages.github.com/
+[gh-pages]: https://github.com/badges/shields/tree/gh-pages
+
+## Deployment
+
+To set things up for deployment:
+
+1.  Get your SSH key added to the server.
+2.  Clone a fresh copy of the repository, dedicated for deployment.
+    (Not required, but recommended; and lets you use `npm ci` below.)
+3.  Add remotes:
+
+```sh
+git remote add s0 root@s0.shields-server.com:/home/m/shields.git
+git remote add s1 root@s1.shields-server.com:/home/m/shields.git
+git remote add s2 root@s2.shields-server.com:/home/m/shields.git
+```
+
+`origin` should point to GitHub as usual.
+
+4.  Since the deploy uses `git worktree`, make sure you have git 2.5 or later.
+
+To deploy:
+
+1.  Use `git fetch` to obtain a current copy of
+    `local-shields-io-production.yml` from the server (or obtain the current
+    version of that file some other way). Save it in `config/`.
+2.  Check out the commit you want to deploy.
+3.  Run `npm ci`. **This is super important for the frontend build!**
+4.  Run `make deploy-s0` to make a canary deploy.
+5.  Check the canary deploy:
+    - [Visit the server][s0]. Don't forget that most of the preview badges
+      are static!
+    - Look for errors in [Sentry][].
+    - Keep an eye on the [status page][].
+6.  After a little while (usually 10–60 minutes), finish the deploy:
+    `make push-s1 push-s2 deploy-gh-pages`.
+
+To roll back, check out the commit you want to roll back to and repeat those
+steps.
+
+To see which commit is deployed to a server run `git ls-remote` and then
+`git log` on the `HEAD` ref. There will be two deploy commits preceded by the
+commit which was deployed.
+
+Be careful not to push the deploy commits to GitHub.
+
+`make deploy-s0` does the following:
+
+1.  Creates a working tree in `/tmp`.
+2.  In that tree, runs `features` and `examples` to generate data files
+    needed for the frontend.
+3.  Builds and checks in the built frontend.
+4.  Checks in `local-shields-io-production.yml`.
+5.  Pushes to s0, which updates dependencies and then restarts itself.
+
+`make push-s1 push-s2 deploy-gh-pages` does the following:
+
+1.  Pushes the same working tree to s1 and s2.
+2.  Creates a new working tree for the frontend.
+3.  Adds a commit cleaning out the index.
+4.  Adds another commit with the build frontend.
+5.  Pushes to `gh-pages`.
+
+## DNS
+
+I'm not sure where the DNS is registered. @olivierlacan has access.
+
+## Logs
+
+Logs are available on the individual servers via SSH. @espadrine has access.
+
+## Error reporting
+
+[Error reporting][sentry] is one of the most useful tools we have for monitoring
+the server. It's generously donated by [Sentry][sentry home]. We bundle
+[`raven`][] into the application, and the Sentry DSN is configured via `local- shields-io-production.yml` (see [documentation][sentry configuration]).
```

Comment:
raven link not working here (think it is because of the `` )

Reply:
Fixed!